gslib/utils/stet_util.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for Split Trust Encryption Tool integration."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+
+import os
+import shutil
+
+from gslib import storage_url
+from gslib.utils import execution_util
+from gslib.utils import temporary_file_util
+
+from boto import config
+
+DEFAULT_STET_CONFIG_PATH = '~/.config/stet.yaml'
+
+
+class StetSubcommandName(object):
+  """Enum class for available STET subcommands."""
+  ENCRYPT = 'encrypt'
+  DECRYPT = 'decrypt'
+
+
+def _get_stet_binary_from_path():
```

Comment:
I think this is not needed. Subprocess.Popen should be able to find the correct binary. Just doing
```
subprocess.Popen(['stet', ....]) should be enough.
```

Reply:
Yes, but because we catch the missing binary, we can give a clearer error.
Otherwise, you get something similar to a `cp` upload source file missing.
```
>>> import subprocess
>>> subprocess.Popen('missing_stet')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.9/subprocess.py", line 1823, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'missing_stet'
```